configure

```diff
@@ -1252,15 +1252,30 @@ case "${ARCH}" in
             ARCH_SHARED_OBJS="${ARCH_SHARED_OBJS} armfeature.lo"
         fi
 
+        cat > $test.c << EOF
+int main() { return 0; }
+EOF
         GCC_MACHINE=$(${CC} -dumpmachine)
         case "${GCC_MACHINE}" in
             *gnueabihf)
-                floatabi="-mfloat-abi=hard" ;;
+                if try $CC -c $CFLAGS $test.c -mfloat-abi=hard; then
+                  floatabi="-mfloat-abi=hard"
+                fi
+            ;;
             *gnueabi)
-                floatabi="-mfloat-abi=softfp" ;;
+                if try $CC -c $CFLAGS $test.c -mfloat-abi=softfp; then
+                  floatabi="-mfloat-abi=softfp"
+                fi
+            ;;
         esac
-        CFLAGS="${CFLAGS} ${floatabi}"
-        SFLAGS="${SFLAGS} ${floatabi}"
+        if [ -z $floatabi ]; then
+          echo "ARM floating point arch not auto-detected" | tee -a configure.log
```

Comment:
Perhaps this should be a warning/error instead?
Is this a "valid" state (assuming 100% working soft/hard-float detection)?

Reply:
I would rather it just proceed normally instead of full stop but I'm not an ARM expert.

Reply:
Earlier I was thinking it should try both versions if the last part of the triplet doesn't mention which one the compiler should support and then finally revert to having NEON optimizations disabled.

Last time I checked in Linaro toolkit for ARM, compiler that only supports either soft or hard float, has only one version of the internal headers, but it is possible to have compiler that has both versions of internal headers.

Most newer processors should use soft float, but older processors might work with hard float. NEON is basically overlapping set of older vector instruction set which is supported by original ARM compiler for Visual C++.

Reply:
I could add an additional check for hard float-abi when it is `.*gnueabi` but do we even want to be doing detection based on triplet? We can just check for both and the first one it finds then it uses it.

Reply:
As far as I know having "hf" in the triplet is kind of Debianism... Not depending on it makes it possible to support other Linux versions better... As I said, hard float is more common on ARMv7 and soft float is more common on ARMv8. We (should) support ARMv6 upwards when vector optimizations are enabled.

Reply:
I have removed the triplet check.